gcc/rust/backend/rust-compile-type.cc

```diff
@@ -532,25 +532,19 @@ TyTyResolveCompile::visit (const TyTy::DynamicObjectType &type)
 
   tree uint = ctx->get_backend ()->integer_type (
     true, ctx->get_backend ()->get_pointer_size ());
-  tree uintptr_ty = ctx->get_backend ()->pointer_type (uint);
+  tree uintptr_ty = build_pointer_type (uint);
 
-  Backend::typed_identifier f ("__receiver_trait_obj_ptr", uintptr_ty,
+  Backend::typed_identifier f ("__ptr", uintptr_ty,
 			       ctx->get_mappings ()->lookup_location (
 				 type.get_ty_ref ()));
   fields.push_back (std::move (f));
 
-  for (size_t i = 0; i < items.size (); i++)
-    {
-      // mrustc seems to make a vtable consisting of uintptr's
-      tree uint = ctx->get_backend ()->integer_type (
-	true, ctx->get_backend ()->get_pointer_size ());
-      tree uintptr_ty = ctx->get_backend ()->pointer_type (uint);
-
-      Backend::typed_identifier f ("__" + std::to_string (i), uintptr_ty,
-				   ctx->get_mappings ()->lookup_location (
-				     type.get_ty_ref ()));
-      fields.push_back (std::move (f));
-    }
+  tree vtable_size = build_int_cst (size_type_node, items.size ());
+  tree vtable_type = ctx->get_backend ()->array_type (uintptr_ty, vtable_size);
+  Backend::typed_identifier vtf ("__vtptr", vtable_type,
```

Comment:
Can we run into any issue where the identifier created here would also be a valid "user" identifier? Would it make more sense to instead use characters that the user could never input in a valid identifier? Something like `@vtptr`, `--vtptr` or `#vtptr` etc

Reply:
Yeah i know what you mean in C++ they actually use the create_tmp_var code to generate a prefix record with a unique identifier which is a neat trick.

But in rustic they actually don't do that: https://github.com/Rust-GCC/gccrs/issues/854

They actually seem to make a structure like: struct dyn{ pointer , vtable }. So to get the gdb integration i should actually be using those field names instead.